### PR TITLE
filter: Add .md docs

### PR DESCRIPTION
**Pull request by toddmgreer**
Description: Add docs as .md files in CacheFilter's source dir until we're ready for deployment.
Risk Level: Low (Nothing references these files)
Testing: n/a
Docs Changes: Nothing in the published docs; nothing points to these
Release Notes: n/a
#868

### CODE REVIEW

File: source/extensions/filters/http/cache/cache_filter.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+### HTTP Cache Filter
```
Comment (mattklein123) [inline source/extensions/filters/http/cache/cache_filter.md]: Right now all of our dev docs are in /source/docs for better worse. WDYT about putting this in /source/docs/filters/http/cache/... or something like that? @jmarantz any thoughts on this? I'm fine either way. In general if we thing dev docs should live closer to the code they talk about maybe we should move the other docs as a separate thing. I could go either way on this.

Comment (toddmgreer) [reply source/extensions/filters/http/cache/cache_filter.md]: Done. (I think I misunderstood an earlier suggestion.)

**Pull request merged**